### ISSUE TITLE
Physics override not working properly when speed is set to 0 while moving

### ISSUE THREAD

**Issue by Zughy** [luanti-org/luanti#10037]
##### Minetest version

```
5.2.0
```

##### Summary

Trying to set 
```
player:set_physics_override({
              speed = 0,
              })
```
doesn't work properly while a movement key is held down (players slide in the last direction they were moving to). I've tried with set_velocity and set_acceleration equal to {x=0,y=0,z=0} but again no luck

##### Steps to reproduce
Overrides physics to 0 while moving

**Comment by ClobberXD** [comment]
Is this just how it appears to the client, or does the server actually register the player as moving? I'm betting on the former, as you mention `set_velocity` doesn't have an effect.

**Comment by oilboi** [comment]
I noticed this, I'm not sure why it acts so crazy

**Comment by sfan5** [comment]
The reason for this is that setting speed to zero not only prevents acceleration but also deceleration ([parameters here](https://github.com/minetest/minetest/blob/0a1181f7637adab6fe1ba5d3c09038912a06b37f/src/client/localplayer.cpp#L673)).

> I've tried with set_velocity and set_acceleration equal to {x=0,y=0,z=0} but again no luck

Note the these methods do nothing for players.

**Comment by appgurueu** [comment]
IMO not a bug as velocity should be kept. You can use `player:add_velocity(vector.multiply(player:get_player_velocity(), -1))` BTW.

**Comment by jastevenson303** [comment]
I'm inclined to agree.  However, when sleeping for example, your speed is
set to zero.  However, now that a punching player can apply velocity to
your character, there's no guarantee you'll stay situated sleeping in the
bed.  Just food for thought.

**Comment by Zughy** [comment]
> IMO not a bug as velocity should be kept. You can use `player:add_velocity(vector.multiply(player:get_player_velocity(), -1))` BTW.

I don't agree as it's pretty unintuitive. I'd expect to have the player unable to move if I set speed = 0

**Comment by hecktest** [comment]
set/add velocity does not work for players.
Using the movement speed as a friction coefficient is a bug.

**Comment by appgurueu** [comment]
> set/add velocity does not work for players.

It does (used to be called `add_player_velocity` though), only possible problem might be desync. As a workaround, you should probably attach the player to an invisible entity.

> Using the movement speed as a friction coefficient is a bug.

true

**Comment by hecktest** [comment]
Oh, sorry, I was looking at outdated (0.4) documentation. Back in the day there was truly no way to manipulate player velocity, which was pretty frustrating.

**Comment by appgurueu** [comment]
> Oh, sorry, I was looking at outdated (0.4) documentation. Back in the day there was truly no way to manipulate player velocity, which was pretty frustrating.

Back in the day modders would temporarily attach players to entities to move them :'-)

**Comment by grorp** [comment]
For backwards compatibility, we can't change the behavior of the `speed` physics override so that it doesn't affect acceleration (and deceleration).

Fortunately, with #11465 already merged and #13815 hopefully coming soon, you will be able to control all speed values separetely (using `speed_walk`,  `speed_climb`, `speed_crouch` and `speed_fast`). These new overrides don't affect acceleration, so this issue can be solved by replacing

```lua
player:set_physics_override({
    speed = 0,
})
```

with

```lua
player:set_physics_override({
    speed = 1,
    speed_walk = 0,
    speed_climb = 0,
    speed_crouch = 0,
    speed_fast = 0,
    -- The following is optional and causes the player to stop immediately
    -- instead of coming to a halt more slowly.
    acceleration_default = math.huge,
    acceleration_air = math.huge,
    acceleration_fast = math.huge,
})
```

-------

However, #13796 could still be an option to make the behavior of `speed` more reasonable.

**Comment by Zughy** [comment]
@grorp yes, #13796 is way easier. What you've suggested sounds like a messy hack (no offence)
